photon-client/src/main/scala/com/linkedin/photon/ml/cli/game/scoring/Params.scala

```diff
@@ -242,6 +251,16 @@ object Params {
             "convention.")
         .foreach(x => params.offHeapIndexMapNumPartitions = x)
 
+      opt[String]("input-columns-names")
+        .text(s"A map of the input columns names, in format: response:label|offset:offset|uid:uid ... ")
+        .foreach(x =>
+          x.split("\\|")
+            .foreach { line =>
+              val Array(key, value) = line.split(":").map(_.trim)
+              params.inputColumnsNames.updated(InputColumnsNames.withName(key), value)
+            }
+        )
+
```

Comment:
Even if we have those `set` methods on the `GameEstimator`, we don't have to expose that to the user with 5 distinct parameters. We can still have a user API with a "convenient" way to set some or all of the column names, ship those in a `Map` to the `Driver` and have the `Driver` call these setters with the contents of the `Map`. That reduces our internal code clutter, while still using spark.ml-style setters on `GameEstimator`.